app/javascript/psulib_blacklight/javascripts/availability.js

```diff
@@ -228,50 +257,73 @@ function groupByLibrary(holdings) {
-                statusCode = statusText = 'Checked Out';
-                break;
-            case 'MISSING':
-                statusCode = statusText = 'Missing';
-                break;
-            case 'ILLEND':
-                statusCode = 'ILLEND';
-                statusText = `<a data-type="ill-link" data-catkey="${titleID}" href="#"><span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span> This copy unavailable, submit request via Interlibrary Loan</a>`;
-                break;
-            default:
-                statusCode = statusText = 'Not Available';
-        }
+function resolveLocation(item, request_via_ill_locations, all_locations) {
+    var location = '';
+    var spinner = '<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>';
+
+    // Check request via ILL locations
+    if (item.currentLocationID in request_via_ill_locations) {
+        location = `<a data-type="ill-link" data-catkey="${item.catkey}" data-call-number="${item.callNumber}" href="#">${spinner}Copy unavailable, request via Interlibrary Loan</a>`;
+    } else if (['ARKTHESES', 'AH-X-TRANS'].includes(item.currentLocationID)) {
+        var aeonLocation = (item.currentLocationID in all_locations) ? all_locations[item.currentLocationID] : "";
+        var shared = `data-catkey="${item.catkey}" data-call-number="${item.callNumber}" data-archival-thesis`;
+
+        location = `<a data-type="ill-link" ${shared} href="#">${spinner}Request Scan - Penn State Users</a><br>
+                    <a href="https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html">Request Scan - Guest</a><br>
+                    <a data-type="aeon-link" ${shared} data-item-id="${item.itemID}" data-item-location="${aeonLocation}" href="#">${spinner}View in Special Collections</a>`;
+    } else {
+        location = (item.currentLocationID in all_locations) ? all_locations[item.currentLocationID] : "";
     }
 
-    return {statusCode:statusCode, statusText: statusText};
+    return location;
 }
 
-function createILLURL(jQueryObj, catkey) {
-    $.get(`/catalog/${catkey}/raw.json`, function(data) {
-        var ILLURL = "https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll?Action=10&Form=30";
+function createILLURL(jQueryObj, item) {
+    $.get(`/catalog/${item.catkey}/raw.json`, function(data) {
+        var ILLURL = "https://psu-illiad-oclc-org.ezaccess.libraries.psu.edu/illiad/upm/illiad.dll/OpenURL?Action=10";
+
         if (Object.keys(data).length > 0) {
-            var ISBN = data.isbn_ssm;
-            var title = data.title_display_ssm;
-            var author = data.author_tsim;
-            var pubdate = data.pub_date_itsi;
-            ILLURL += `&isbn=${ISBN}&title=${title}&aulast=${author}.&rfr_id=info%3Asid%2Fcatalog.libraries.psu.edu&date=%5B${pubdate}%5D`;
+            var title = data.title_245ab_tsim;
+            var author = data.author_tsim ? data.author_tsim : "";
```

Comment:
I guess this is necessary eh? If `data.author_tsim` is missing does it break things?

Reply:
yeah it was passing it as `undefined`

Reply:
Nice catch, thanks.